### PR TITLE
EIP191 Personal Signing and EIP4361

### PR DESCRIPTION
**Pull request by dospore**
Initial commit for the implementation of EIP191 and EIP4361 mentioned in #513 and #514 

Extending signTypedData implementation to support personal signatures.

Signing:
image

Test Account:
image

### CODE REVIEW

File: background/main.ts [hunk 1]
```diff
@@ -268,9 +271,10 @@ export default class Main extends BaseService<never> {
       ? (Promise.resolve(null) as unknown as Promise<LedgerService>)
       : LedgerService.create()
 
-    const signingService = HIDE_IMPORT_LEDGER
-      ? (Promise.resolve(null) as unknown as Promise<SigningService>)
-      : SigningService.create(keyringService, ledgerService, chainService)
+    // const signingService = HIDE_IMPORT_LEDGER
+      // ? (Promise.resolve(null) as unknown as Promise<SigningService>)
+      // : SigningService.create(keyringService, ledgerService, chainService)
```
Comment (dospore) [inline background/main.ts:276]: This was just my small brain trying to test

File: background/main.ts [hunk 2]
```diff
@@ -382,10 +386,12 @@ export default class Main extends BaseService<never> {
         handler: () => this.store.dispatch(initializationLoadingTimeHitLimit()),
       },
     })
-
+  
     // Start up the redux store and set it up for proxying.
     this.store = initializeStore(savedReduxState, this)
 
+    console.log('store', this.store)
```
Comment (dospore) [inline background/main.ts:393]: This never seems to load in the initial signing state

File: background/redux-slices/index.ts [hunk 3]
```diff
@@ -23,8 +23,8 @@ const mainReducer = combineReducers({
   transactionConstruction: transactionConstructionReducer,
   ui: uiReducer,
   dappPermission: dappPermissionReducer,
-  ...(HIDE_IMPORT_LEDGER ? {} : { ledger: ledgerReducer }),
   signing: signingReducer,
+  ...(HIDE_IMPORT_LEDGER ? {} : { ledger: ledgerReducer }),
```
Comment (dospore) [inline background/redux-slices/index.ts:27]: was just moving it below

File: background/services/keyring/index.ts [hunk 4]
```diff
@@ -445,6 +445,40 @@ export default class KeyringService extends BaseService<Events> {
...
+  }: {
+    signingData: EIP191Data
+    account: HexString
+  }): Promise<string> {
+    this.requireUnlocked()
+    // const { domain, types, message } = typedData
+    // find the keyring using a linear search
+    const keyring = await this.#findKeyring(account)
+    // When signing we should not include EIP712Domain type
+    // const { EIP712Domain, ...typesForSigning } = types
```
Comment (dospore) [inline background/services/keyring/index.ts:469]: will remove the typedData stuff

**Comment by dospore** [comment]
Should be good to go now, removed the lint errors, there are a tonne of warnings but I dont think they are mine and its probably for another pr

**Comment by dospore** [comment]
Local build works, sorry

**Comment by dospore** [comment]
There are a few fixes included in https://github.com/dospore/extension/pull/1 that I may need to be included before this gets merged in. Mainly the conversion from hex to ascii as currently it signs the hex value which results in the wrong expected result

**Comment by greg-nagy** [comment]
@dospore Thanks for this PR, and sorry for the long wait.

We have an internal goal to reivew this and merge it today or tomorrow. 

To prepare for merge, could you please:
- resolve the conflicts / update the branch to the latest main
- make sure that all the git commits are signed by a verified user? 
  - https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits - here is a doc on how to set it up 
  - and you can use this command after it. has been set up `git rebase --preserve-merges --exec 'git commit --amend --no-edit -n -S' -i origin/main` this command can help you 
  
  Thank you 🙏

**Review by greg-nagy — APPROVED** [review]
Codewise I am really impressed by this PR. 

Most of the changes at wiring is correct, which is not a small feat in this part of the codebase. 

The feature won't work with ledger, but I think this worth mergeing as is and make the updates later based on the feedbacks. 

None of the comments are blocking. 

Also tested the feature with chrome on login.xyz and it worked like a charm.

**Review by greg-nagy — APPROVED** [review]
it would seem that this is ready to 🚢

**Pull request merged**